File: android/src/main/java/com/iconshot/detonator/emitter/EventEmitter.java
package com.iconshot.detonator.emitter;

import com.iconshot.detonator.Detonator;

public class EventEmitter extends Emitter {
    public EventEmitter(Detonator detonator) {
        super(detonator);
    }

    public void emit(String name) {
        emit(name, null);
    }

    public void emit(String name, Object data) {
        Event handler = new Event();

        handler.name = name;
        handler.data = data;

        detonator.emit("event", handler);
    }

    private static class Event {
        String name;
        Object data;
    }
}

File: ios/Sources/Detonator/Emitter/EventEmitter.swift
public class EventEmitter: Emitter {
    public func emit(name: String, data: Encodable?) {
        let handler = Event(name: name, data: data)
        
        detonator.emit(name: "event", value: handler)
    }
    
    struct Event: Encodable {
        let name: String
        let data: AnyEncodable?

        init(name: String, data: Encodable?) {
            self.name = name
            
            if let data = data {
                self.data = AnyEncodable(data)
            } else {
                self.data = nil
            }
        }
    }
}
